packages/taquito/test/contract/rpc-contract-provider.spec.ts

```diff
@@ -1235,6 +1237,208 @@ describe('RpcContractProvider test', () => {
     });
   });
 
+  describe("transferTicket", () => {
+    it("validate that a reveal operation will be added when needed", async (done) => {
+      mockRpcClient.getBlockMetadata.mockResolvedValue(TransferTicketNoReveal)
+      const params: TransferTicketParams = {
+        source: 'tz1iedjFYksExq8snZK9MNo4AvXHBdXfTsGX',
+        fee: 804,
+        gasLimit: 5009,
+        storageLimit: 130,
+        ticketContents: { "string": "foobar" },
+        ticketTy: { "prim": "string" },
+        ticketTicketer: 'KT1AL8we1Bfajn2M7i3gQM5PJEuyD36sXaYb',
+        ticketAmount: 2,
+        destination: 'KT1SUT2TBFPCknkBxLqM5eJZKoYVY6mB26Fg',
+        entrypoint: 'default',
+        };
+      const result = await rpcContractProvider.transferTicket(params)
+      const expectedReveal = {
+          counter: "1",
+          fee: "374",
+          gas_limit: "1100",
+          kind: "reveal",
+          public_key: "test_pub_key",
+          source: "test_pub_key_hash",
+          storage_limit: "0",
+        }
+      const expecteReturn = {
+        counter: "2",
+        destination: "KT1SUT2TBFPCknkBxLqM5eJZKoYVY6mB26Fg",
+        entrypoint: "default",
+        fee: "804",
+        gas_limit: "5009",
+        kind: "transfer_ticket",
+        source: "tz1iedjFYksExq8snZK9MNo4AvXHBdXfTsGX",
+        storage_limit: "130",
+        ticket_amount: "2",
+        ticket_contents: {
+          string: "foobar",
+        },
+        ticket_ticketer: "KT1AL8we1Bfajn2M7i3gQM5PJEuyD36sXaYb",
+        ticket_ty: {
+          prim: "string",
+        }
+      }
+      const actual = result.raw.opOb.contents ?? []
+      expect(actual[0]).toEqual(expectedReveal)
+      expect(actual[1]).toEqual(expecteReturn)
+      expect
+      done()
+    })
+    it("validate that a reveal option wont be added when not needed", async (done) => {
+      mockRpcClient.getManagerKey.mockReturnValue(null);
```

Comment:
getManagerKey should not be null here. We want to test that if the account is already revealed taquito won't add a reveal operation. (When an account is reveled, it has a manager)